### ISSUE TITLE
do not auto-add save_result when export_workspace is used

### ISSUE THREAD

**Issue by jdries** [Open-EO/openeo-python-client#623]

**Comment by soxofaan** [comment]
Indeed, there is quite some technical debt to clean up around "save_result" handling. Related tickets:
- https://github.com/Open-EO/openeo-python-client/issues/583
- https://github.com/Open-EO/openeo-python-client/issues/402
- https://github.com/Open-EO/openeo-python-client/issues/401
- https://github.com/Open-EO/openeo-python-client/issues/391

**Comment by soxofaan** [comment]
started working on draft at #630

**Comment by soxofaan** [comment]
done with 58f0205db63efbc860b99477cc5e5dea6069224b
